File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        if ( m*n != original.length) {
            return new int[0][0];
        }
        int[][] answer = new int[m][n];
        int row = 0;
        int col = 0;
        for (int element: original) {
            answer[row][col] = element;
            col++;
            if (col == n) {
                col = 0;
                row++;
            }
        }
        return answer;
    }
}

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.go
func construct2DArray(original []int, m int, n int) [][]int {
	ans := make([][]int, m)
    if m*n != len(original) {
		return make([][]int, 0)
	}
	row := 0
	col := 0
	for _, v := range original {
		if col == 0 {
			ans[row] = make([]int, n)
		}
		ans[row][col] = v
		col++
		if col == n {
			row++
			col = 0
		}
	}
	return ans
}